pkg/controller/storagecluster/initialization_reconciler.go

```diff
@@ -55,9 +80,120 @@ func (r *ReconcileStorageCluster) ensureStorageClasses(instance *ocsv1.StorageCl
 			}
 		}
 	}
+	return nil
+}
 
-	instance.Status.StorageClassesCreated = true
-
+// ensureExternalCluster ensures that Storagclass for the external cluster exist
+func (r *ReconcileStorageCluster) ensureExternalCluster(instance *ocsv1.StorageCluster, reqLogger logr.Logger) error {
+	// check for the status boolean value accepted or not
+	if instance.Status.ExternalSecretFound {
+		return nil
+	}
+	found := &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      externalClusterDetailsSecret,
+			Namespace: instance.Namespace,
+		},
+	}
+	err := r.client.Get(context.TODO(), types.NamespacedName{Name: found.Name, Namespace: found.Namespace}, found)
+	if err != nil {
+		return err
+	}
+	ownerRef := metav1.OwnerReference{
+		UID:        instance.UID,
+		APIVersion: instance.APIVersion,
+		Kind:       instance.Kind,
+		Name:       instance.Name,
+	}
+	var data []ExternalResource
+	err = json.Unmarshal(found.Data[externalClusterDetailsKey], &data)
+	if err != nil {
+		reqLogger.Error(err, "could not parse json blob")
+		return err
+	}
+	scs, err := r.newStorageClasses(instance)
+	if err != nil {
+		return err
+	}
+	// newStorageClass might not return OBC storageClass so checking the
+	// number of storageClasses returned
+	if len(scs) < 3 {
+		scs = append(scs, r.newOBCStorageClass(instance))
+	}
```

Comment:
Don't do this. If you can't rely on `newStorageClasses()` to create the OBC class, fix it.

Reply:
Done, fixed it by adding `initData.Spec.ExternalStorage.Enable` condition.